packages/medusa/src/repositories/product-category.ts

```diff
@@ -1,20 +1,33 @@
 import { EntityRepository, TreeRepository, Brackets, ILike } from "typeorm"
 import { ProductCategory } from "../models/product-category"
-import { ExtendedFindConfig, Selector } from "../types/common"
+import { ExtendedFindConfig, Selector, QuerySelector } from "../types/common"
 
 @EntityRepository(ProductCategory)
 export class ProductCategoryRepository extends TreeRepository<ProductCategory> {
   public async getFreeTextSearchResultsAndCount(
     options: ExtendedFindConfig<ProductCategory, Selector<ProductCategory>> = {
       where: {},
     },
-    q: string | undefined
+    q: string | undefined,
+    treeSelector: QuerySelector<ProductCategory> = {},
```

Comment:
Need some comments on this one:

Goal:
- To be able to filter children and parent tree nodes with **default scope**
- To be able to filter only on category items (default scope + filter params)